content/en/guide/v10/refs.md

```diff
@@ -1,124 +1,215 @@
 ---
 name: References
-description: 'References can be used to access raw DOM nodes that Preact has rendered'
+description: 'Refs are a way of creating stable values that are local to a component instance and persist across renders.'
 ---
 
 # References
 
-There will always be scenarios where you need a direct reference to the DOM-Element or Component that was rendered by Preact. Refs allow you to do just that.
+References, or refs for short, are stable, local values that persist across component renders but don't trigger rerenders like state or props would when they change.
 
-A typical use case for it is measuring the actual size of a DOM node. While it's possible to get the reference to the component instance via `ref` we don't generally recommend it. It will create a hard coupling between a parent and a child which breaks the composability of the component model. In most cases it's more natural to just pass the callback as a prop instead of trying to call the method of a class component directly.
+Most often you'll see refs used to facilitate imperative manipulation of the DOM but they can be used to store any arbitrary local value that you need to be kept stable. You may use them to track a previous state value, keep a reference to an interval or timeout ID, or simply a counter value. Importantly, refs should not be used for rendering logic, instead, consumed in lifecycle methods and event handlers only.
 
 ---
 
 <div><toc></toc></div>
 
 ---
 
-## createRef
+## Creating a Ref
 
-The `createRef` function will return a plain object with just one property: `current`. Whenever the `render` method is called, Preact will assign the DOM node or component to `current`.
+There are two ways to create refs in Preact, depending on your preferred component style: `createRef` (class components) and `useRef` (function components/hooks). Both APIs fundamentally work the same way: they create a stable, plain object with a `current` property, optionally initialized to a value.
+
+```jsx
+// Class components
+import { createRef } from "preact";
+
+class MyComponent extends Component {
+  countRef = createRef();
+  inputRef = createRef(null);
+
+  // ...
+}
+```
+
+```jsx
+// Function components
+import { useRef } from "preact/hooks";
+
+function MyComponent() {
+  const countRef = useRef();
+  const inputRef = useRef(null);
+
+  // ...
+}
+```
```

Comment:
I'm thinking of adding a toggle widget to switch between class examples & hooks, might make this a bit cleaner.

Still have to think of a way to do that which doesn't break translated pages that might be out of sync though.